src/project.rs

```diff
@@ -97,17 +97,28 @@ pub struct Project {
     pub stmts: Vec<ast::Stmt>,
 }
 
-pub fn save<P: AsRef<Path>>(path: P, project: Project) -> Result<(), ProjectError> {
+/// Saves project to given path. If this path does not contain valid project
+/// extension, it is automatically added.
+///
+/// Returns `PathBuf` which can be different than original path if the project
+/// extension was added.
+pub fn save<P: AsRef<Path>>(path: P, project: Project) -> Result<PathBuf, ProjectError> {
     let pretty_config = ron::ser::PrettyConfig::default();
     let mut serializer = ron::ser::Serializer::new(Some(pretty_config), true);
     project.serialize(&mut serializer)?;
 
+    let mut path_buf = path.as_ref().to_path_buf();
+
+    if path_buf.extension().is_none() {
+        path_buf.set_extension(PROJECT_EXTENSION);
+    }
```

Comment:
Should we be assuming that if there is an extension, it is correct one? The doc comment says "If this path does not contain valid project extension", but this code doesn't handle validity.

Reply:
It's fixed now. If you try to save `gurban.turban` it actually saves `gurban.turban.hurban`. @janper please test this fix one more time as well.